src/login.ts

```diff
@@ -6,52 +6,94 @@ import { MastoConfig } from './config';
 import { HttpNativeImpl } from './http';
 import type { LogType } from './logger';
 import { LoggerConsoleImpl } from './logger';
+import type { v1, v2 } from './mastodon';
 import { Client } from './mastodon';
-import { InstanceRepository } from './mastodon/v1/repositories';
+import { InstanceRepository as V1InstanceRepository } from './mastodon/v1/repositories';
+import { InstanceRepository as V2InstanceRepository } from './mastodon/v2/repositories';
 import { SerializerNativeImpl } from './serializers';
 import type { Writable } from './utils';
 import { WsNativeImpl } from './ws';
 
-export type LoginParams = {
+export interface RequestParams {
   /** URL of your instance */
   readonly url: string;
-  /** Log level to print to your console */
-  readonly logLevel?: LogType;
-  /** Access token of your account */
-  readonly accessToken?: string;
   /** Timeout in milliseconds */
   readonly timeout?: number;
   readonly defaultRequestInit?: Omit<RequestInit, 'body' | 'method'>;
-  readonly disableVersionCheck?: boolean;
   readonly disableDeprecatedWarning?: boolean;
-};
+}
 
-export const login = async (params: LoginParams): Promise<Client> => {
-  const draft: Writable<MastoConfigProps> = {
+const buildConfigProps = (params: RequestParams): MastoConfigProps => {
+  return {
     url: params.url,
     streamingApiUrl: '',
-    logLevel: params.logLevel,
-    accessToken: params.accessToken,
     timeout: params.timeout,
     defaultRequestInit: params.defaultRequestInit,
-    disableVersionCheck: params.disableVersionCheck,
     disableDeprecatedWarning: params.disableDeprecatedWarning,
   };
+};
+
+const buildRequestContext = (
+  props: MastoConfigProps,
+): {
+  serializer: SerializerNativeImpl;
+  config: MastoConfig;
+  http: HttpNativeImpl;
+} => {
   const serializer = new SerializerNativeImpl();
+  const config = new MastoConfig(props, serializer);
+  const http = new HttpNativeImpl(serializer, config);
+  return { serializer, config, http };
+};
 
-  {
-    const config = new MastoConfig(draft, serializer);
-    const http = new HttpNativeImpl(serializer, config);
-    const instance = await new InstanceRepository(http, config).fetch();
-    draft.version = new SemVer(instance.version);
-    draft.streamingApiUrl = instance.urls.streamingApi;
-  }
+export const fetchV1Instance = (
```

Comment:
I added two new functions `fetchV1Instance` and `fetchV2Instance`. Users can use them to get v1 or v2 instance info.